SRM 325/RGBStreet.cpp

```diff
@@ -0,0 +1,196 @@
+#include <cstdio>
+#include <cmath>
+#include <cstring>
+#include <ctime>
+#include <iostream>
+#include <algorithm>
+#include <set>
+#include <vector>
+#include <sstream>
+#include <typeinfo>
+#include <fstream>
+
+using namespace std;
+
+int dp[100];
+
+class RGBStreet {
+    public:
+
+    int cvt_strto_int(string s) {
+        int numb = 0;
+        int n =  s.length();
+        for(int i=0; i<n; ++i) {
+            numb = numb*10 + (s[i]-'0');
+        }
+        return numb;
+    }
+
+    vector<vector<int> > extractPrices(vector<string> houses) {
+        vector<vector<int> > prices(houses.size());
+        for(int i=0 ;i<houses.size(); ++i) {
+            int start = 0;
+            vector<int> tmp;
+            for(int j=0; j<houses[i].length(); ++j) {
+                if(houses[i][j] == ' ') {
+                    string str1 = houses[i].substr(start, j-start);
+                    tmp.push_back(cvt_strto_int(str1));
+                    start = j+1;  
+                }
+                else if(j+1 == houses[i].length()) {
+                    string str1 = houses[i].substr(start);
+                    tmp.push_back(cvt_strto_int(str1));
+                }
+            }
+            prices[i] = tmp;
+        }
+        return prices;
+    }
+
+    int f(int i, int& last, char colour, vector<string>& houses, vector< vector< int> >& prices) {
+        int &res = dp[i];
```

Comment:
There are 2 states -- current index and what was the last character you kept. So, its a 2D dp.